packages/widgets/src/blocks/legacy-widget/edit/utils.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, useLayoutEffect, useState } from '@wordpress/element';
+import { useResizeObserver } from '@wordpress/compose';
+
+/**
+ * Positions an element by the top of an objective element if it can fit or
+ * otherwise as close as possible within the window.
+ *
+ * @param {Element} objective    Element to take position from.
+ * @param {Object}  $1           Options.
+ * @param {boolean} $1.isEnabled Whether to apply the hook or not.
+ *
+ * @return {Object} A ref and resize observer element.
+ */
+export function useBudgeTopBy( objective, { isEnabled } ) {
+	const subjectRef = useRef();
+	const objectiveRef = useRef();
```

Comment:
Another naming thing, I'm finding these aren't clear enough. I would regularly confuse the subject and the objective.

I think like other popover positioning utilities, it'd be good to stick with the term `anchor` for the element that the popover 'achors' to.

`target` is a term I find used more than objective, but could also call it `popoverRef` to use simpler language.

Reply:
I've renamed `subjectRef` to `ref` (it's the one returned by the hook for attachment on the element to be positioned) and `objectiveRef` is now `anchorRef`.